doc/src/manual/parallel-computing.md

```diff
@@ -1,12 +1,542 @@
 # Parallel Computing
 
-This part of the manual details the following types of parallel programming available in Julia.
+For any newcomer to multi-thread and parallel computing it can be useful to first appreciate 
+the different levels of parallelism offered by Julia. We can divide them in three main categories :
 
-1. Distributed memory using multiple processes on one or more nodes
-2. Shared memory using multiple processes on a single node
-3. Multi-threading
+1. Julia Coroutines (Green Threading)
+2. Multi-Threading
+3. Multi-Core or Distributed Processing
 
-# Distributed Memory Parallelism
+We will consider first Julia [Tasks (aka Coroutines)](@ref man-tasks) and other high-level interfaces and
+modules that rely on the Julia runtime library, that allow to suspend and resume computations with full
+control of inter-`Tasks` communication without worrying of the operative system's scheduler. Julia also
+allows to communicate between `Tasks` through operations like [`wait`](@ref) and [`fetch`](@ref).
+Communication and data synchronization is managed through [`Channel`](@ref)s, which are the conduit
+that allows inter-`Tasks` communication.
+
+Julia also supports multi-threading, where execution is forked and an anonymous function is run across all
+threads. Described as a fork-join approach, all threads have to join together and serial execution continues.
+Multi-threading is supported using the `Base.Threads` module that is still considered experimental, as Julia is
+not fully thread-safe, consider for example [this issue](https://github.com/JuliaLang/julia/issues/22581).
+Also multi-threading should be implemented only if you take into consideration global variables, locks and
+atomics, so we will present it later.
+
+In the end we will present Julia's way to distributed and parallel computing. With scientific computing
+in mind, Julia natively implements interfaces to send a process through multiple cores or machines.
+Also we will mention useful external packages for distributed programming like `MPI.jl` and `DistributeArrays.jl`.
+
+# Coroutines
+
+Julia's parallel programming platform uses [Tasks (aka Coroutines)](@ref man-tasks) to switch among multiple
+computations. Whenever code performs a communication operation like [`fetch`](@ref) or [`wait`](@ref),
+the current task is suspended and a scheduler picks another task to run. A task is restarted when
+the event it is waiting for completes.
+
+For many problems, it is not necessary to think about tasks directly. However, they can be used
+to wait for multiple events at the same time, which provides for *dynamic scheduling*. In dynamic
+scheduling, a program decides what to compute or where to compute it based on when other jobs
+finish. This is needed for unpredictable or unbalanced workloads, where we want to assign more
+work to processes only when they finish their current tasks.
+
+The [`@spawn`](@ref) macro allows to execute an expression on a remote node, choosen automatically,
```

Comment:
I still don't like that this basically discusses the distributed programming model, before introducing it formally.

Maybe an example involving IO or take inspiration form the examples from goroutines?

Reply:
Sorry, after removing the pmap example I forgot to remove this. spawn macros are presented in the Distributed Processing section. 
The paragraph on `remotecall_fetch` and `nextidx` has been moved to the Distrubuted Processing paragraph.